pkg/asset/agent/image/oc.go

```diff
@@ -182,14 +182,21 @@ func (r *release) extractFileFromImage(image, file, cacheDir string) (string, er
 	} else {
 		cmd = fmt.Sprintf(templateImageExtract, file, cacheDir, image)
 	}
+	// Remove file if it exists
+	path := filepath.Join(cacheDir, path.Base(file))
+	os.Remove(path)
```

Comment:
We should definitely ignore `fs.ErrNotExist`, but it would be better if we handled other errors. e.g. a permissions problem here could result in us using an old ISO if the release payload has only the wrong arch.